### PR TITLE
[MISC] use new CPO-way to define seqan3::max_pseudoknot_depth

### PR DESCRIPTION
**Pull request by marehr**
Part of https://github.com/seqan/product_backlog/issues/298.

### CODE REVIEW

File: include/seqan3/alphabet/structure/concept.hpp [hunk 1]
```diff
@@ -267,46 +267,66 @@ namespace seqan3::detail::adl_only
...
+
+    /*!\brief If `alphabet_type` isn't std::is_nothrow_default_constructible, max_pseudoknot_depth will be called with
+     *        std::type_identity instead of a default constructed alphabet.
+     */
+    template <typename alphabet_type>
+    using alphabet_or_type_identity
+        = std::conditional_t<std::is_nothrow_default_constructible_v<std::remove_cvref_t<alphabet_type>> &&
+                             seqan3::is_constexpr_default_constructible_v<std::remove_cvref_t<alphabet_type>>,
+                             std::remove_cvref_t<alphabet_type>,
+                             std::type_identity<alphabet_type>>;
```
Comment (SGSSGene) [inline include/seqan3/alphabet/structure/concept.hpp:290]: This is a real improvement!

File: include/seqan3/alphabet/structure/concept.hpp [hunk 2]
```diff
@@ -267,46 +267,66 @@ namespace seqan3::detail::adl_only
...
+
+    /*!\brief CPO overload (1. out of 3 checks): explicit customisation via `seqan3::custom::alphabet`
+     * \tparam alphabet_type The type of the alphabet. (Needed to defer instantiation for incomplete types)
+     */
+    template <typename alphabet_type = alphabet_t>
+    static constexpr auto SEQAN3_CPO_OVERLOAD(priority_tag<3>)
```
Comment (SGSSGene) [inline include/seqan3/alphabet/structure/concept.hpp:296]: I think we should keep the priority tags conistent?
```suggestion
    static constexpr auto SEQAN3_CPO_OVERLOAD(priority_tag<2>)
```

File: include/seqan3/alphabet/structure/concept.hpp [hunk 3]
```diff
@@ -267,46 +267,66 @@ namespace seqan3::detail::adl_only
...
+     *        `max_pseudoknot_depth(alphabet_type{})`
+     * \tparam alphabet_type The type of the alphabet. (Needed to defer instantiation for incomplete types)
+     *
+     * \details
+     *
+     * If the alphabet_type isn't std::is_nothrow_default_constructible,
+     * `max_pseudoknot_depth(std::type_identity<alphabet_type>{})` will be called.
+     */
+    template <typename alphabet_type = alphabet_t>
+    static constexpr auto SEQAN3_CPO_OVERLOAD(priority_tag<2>)
```
Comment (SGSSGene) [inline include/seqan3/alphabet/structure/concept.hpp:311]:
```suggestion
    static constexpr auto SEQAN3_CPO_OVERLOAD(priority_tag<1>)
```

**Review by SGSSGene — CHANGES_REQUESTED** [review]

**Review by SGSSGene — APPROVED** [review]
:+1:

**Review by eseiler — APPROVED** [review]

**Pull request merged**